### PR TITLE
Update to 3.1.4 and build with OpenSSL 3 [skip ci]

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Update to 3.1.4 and build with OpenSSL 3.

3.1 seems to be compatible with OpenSSL 3. 3.2 is officially supported. I looked in Fedora and they don't have any OpenSSL 3 patches even if they are using OpenSSL 3: https://src.fedoraproject.org/rpms/ruby/tree/f37.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,15 +8,15 @@ package:
...
+  sha256: a3d55879a0dfab1d7141fdf10d22a07dbf8e5cdc4415da1bde06127d5cc3c7b6
   patches:
     # Patches/work-arounds for passing make check
     - disable-backtrace-with-lines.patch
     - fix-isinf-redefinition-msvc.patch  # [win]
     - 0001-Dont-t-add-SDKROOT-to-vals.patch
 
 build:
-  number: 1
+  number: 2
```
Comment (sumit0190) [inline recipe/meta.yaml]: Do we need to increment the build number even though we are changing the version number?

**Review by sumit0190 — APPROVED** [review]
LGTM (apart from that one comment)

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: Um, nope, that's a mistake.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: Fixed, thanks for spotting that!

**Review by skupr-anaconda — APPROVED** [review]
LGTM but could you update dev_url as https://github.com/ruby/ruby

**Pull request merged**